app/uk/gov/hmrc/cdsreimbursementclaimfrontend/testonly/controllers/SubmitClaimController.scala

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cdsreimbursementclaimfrontend.testonly.controllers
+
+import cats.data.EitherT._
+import cats.implicits._
+import javax.inject.{Inject, Singleton}
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.cdsreimbursementclaim.utils.Logging
+import uk.gov.hmrc.cdsreimbursementclaim.utils.Logging.LoggerOps
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.Error
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.services.SubmitClaimService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class SubmitClaimController @Inject() (eisService: SubmitClaimService)(implicit
+  mcc: MessagesControllerComponents,
+  ec: ExecutionContext
+) extends FrontendController(mcc)
+    with Logging {
+
+  val claim: Action[AnyContent] = Action.async { implicit request =>
+    (request.method match {
+      case "GET"  => rightT[Future, Error](testRequestBody)
+      case "POST" => fromOption[Future](request.body.asJson, Error("Request Body is not Json!"))
+    })
+      .flatMap(eisService.submitClaim)
+      .fold(
+        e => {
+          logger.warn(s"could not submit claim", e)
+          InternalServerError
+        },
+        response => Ok(response)
+      )
+
+  }
+
+  val testRequestBody: JsValue = Json.parse("""{
```

Comment:
This will have to change. I have added the latest schema as of this morning and this does pass the schema validation.

See here: https://jira.tools.tax.service.gov.uk/secure/attachment/313531/TPI05_Request.json